src/components/activities/ActivityCard.jsx

```diff
@@ -37,7 +37,53 @@ class ActivityCard extends Component {
     showLocation: false,
     owner: null,
   };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      description: this.props.description,
+      showMoreLess: false,
+      toggleBtnText: true,
+    };
+  }
+
+  componentDidMount() {
+    this.truncateDescription();
+  }
+
   statuses = ['pending', 'rejected', 'approved', 'in review'];
+
+  /**
+   * @name truncateDescription
+   * @summary Shortens the description and adds the more/less text
+   * @return {void}
+   */
+  truncateDescription = () => {
```

Comment:
Instead of doing this, we could create another component whose job is to do exactly this, i.e have a shorter description that resizes when  the `Show More` is clicked. Single Responsibility of components will help us ensure that a component does only one thing that it is concerned about. Let me know if you need us to go through this.

Reply:
@kimani-ndegwa I think `truncateDescription` could work best as a helper function instead.